### ISSUE TITLE
Event subprocess doesn't start by signal event.

### ISSUE THREAD

**Issue by daisuke-yoshimoto** [Activiti/Activiti#1638]
Event subprocess that has signal start event doesn't start by signal event.

- [ ] 5.x
- [ ] 6.x
- [ ] develop

![bug](https://user-images.githubusercontent.com/16251291/34296425-0e18c5ce-e756-11e7-91cb-dfae0e18ab07.png)

## Cause

Event subscription is not registered when a process instance start.

### 5.x

* The event definition(signal event) is registered for the subprocess not parent process in XML parsing.
  * https://github.com/Activiti/Activiti/blob/5.x/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/parser/handler/SignalEventDefinitionParseHandler.java#L56

* In case of a message event, it is correctly registered in the event definition of the parent process.
  *  https://github.com/Activiti/Activiti/blob/5.x/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/parser/handler/MessageEventDefinitionParseHandler.java#L60

**Comment by c3s4r** [comment]
Are there any plans to fix this issue on Activiti 6 or 7?

**Comment by Salaboy** [comment]
@daisuke-yoshimoto @c3s4r yes, this should be fixed in develop for 7. I bet that @daisuke-yoshimoto is already working on it :)

**Comment by nsecchi-carecloud** [comment]
Hey guys, I have ran into this problem with Activiti 5.22.0.7. Is there a workaround replacing beans or a handler or something to sort this problem out? I think the main issue is that the start signal event is not registering the subscription to the signal.
Unfortunately, updating to the future version of Activiti that contains the fix for this won't happen any time soon, so I was wondering if someone could shed some light on where the problem is and if, possible, a quick and dirty fix for it.

Cheers!
